### PR TITLE
make init pull request

### PR DESCRIPTION
**Pull request by dawkstorm**
make init pull request

### CODE REVIEW

File: Services/CustomsService.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore;
+
+namespace CustomsController.Services
+{
+    public class CustomsService : ICustomsService
+    {
+        private readonly CustomsContext _customContext; // database itself
+
+        public CustomsService(CustomsContext customsContext)
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: summary is missing

File: Controllers/CustomsController.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+using CustomsController.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+
+namespace CustomsController.Controllers;
+
```
Comment (dawkstorm) [inline Controllers/CustomsController.cs:5]: remove unnecessary new line break

File: Controllers/CustomsController.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+/// <summary>
+/// Controller für Zoll
```
Comment (dawkstorm) [inline Controllers/CustomsController.cs]: in english

File: Controllers/CustomsController.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    /// <param name="country2code">Second country's A2 Isocode</param>
+    [HttpGet]
+    [Route("get-customs")]
+    [AllowAnonymous]
+    public async Task<bool> GetCustoms(string country1code, string country2code)
+    {
+        return _customService.GetCustoms(country1code, country2code);
+    }
+    /// <summary>
+    /// Check whether there are customs between countries and postalCodes
```
Comment (dawkstorm) [inline Controllers/CustomsController.cs]: postal codes - space

File: CustomsController.http [hunk 5]
```diff
@@ -0,0 +1,6 @@
+@CustomsController_HostAddress = http://localhost:5080
```
Comment (dawkstorm) [inline CustomsController.http]: delete file

File: Model/CustomsContext.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    }
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        base.OnModelCreating(modelBuilder);
+        modelBuilder.ApplyConfiguration(new CountryConfiguration());
+    }
+}
+/// <summary>
+/// Database model for countries
```
Comment (dawkstorm) [inline Model/CustomsContext.cs]: new file model, class

File: Model/CustomsContext.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+public class CustomsContext : DbContext
+{
+    /// <summary>
+    /// Countries database
+    /// </summary>
+    public DbSet<Country> Countries => Set<Country>();
+    /// <summary>
+    /// Postcodes database
+    /// </summary>
+    public DbSet<Postleizahl> Postleizahlen => Set<Postleizahl>();
```
Comment (dawkstorm) [inline Model/CustomsContext.cs]: In english

File: Model/CustomsContext.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    /// </summary>
+    public int Id { get; set; }
+    /// <summary>
+    /// Country's A2 isocode
+    /// </summary>
+    public string A2Code { get; set; } //e.g. "DE", "PL"
+    /// <summary>
+    /// Is it a member of EUCU
+    /// </summary>
+    public bool isEUCU { get; set; }
```
Comment (dawkstorm) [inline Model/CustomsContext.cs]: IsEUCU

File: Model/CustomsContext.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    /// <param name="isEUCU">Is it a member of EUCU</param>
+    public Country(string A2Code, bool isEUCU)
+    {
+        this.A2Code = A2Code;
+        this.isEUCU = isEUCU;
+    }
+}
+
+
+public class Postleizahl
```
Comment (dawkstorm) [inline Model/CustomsContext.cs]: new file, in english

File: Program.cs [hunk 10]
```diff
@@ -0,0 +1,37 @@
+using CustomsController.Controllers;
+using CustomsController.Services;
+using Microsoft.EntityFrameworkCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
```
Comment (dawkstorm) [inline Program.cs]: remove comment

File: Services/CustomsService.cs [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        /// </summary>
+        /// <param name="customsContext">for logic</param>
+        public CustomsService(CustomsContext customsContext)
+        {
+            _customContext = customsContext;
+        }
+
+        public CustomsService() { }
+
+        public Country AddNewCountry(string A2Code, bool isEUCU)
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: inheritdoc missing

File: Services/CustomsService.cs [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+            {
+                _customContext.Add(country);
+            }
+            _customContext.SaveChanges();
+            return country;
+        }
+
+        public Country RemoveCountry(string A2Code)
+        {
+            var country = _customContext.Countries.FirstOrDefault(c => c.A2Code == A2Code);
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: Country does not exist info give back

File: Controllers/CustomsController.cs (hunk 2 above)
Comment (dawkstorm) [reply Controllers/CustomsController.cs:5]: done

File: Model/Country.cs [hunk 13]
```diff
@@ -0,0 +1,29 @@
+/// <summary>
+/// Database model for countries
+/// </summary>
+public class Country
+{
+
```
Comment (dawkstorm) [inline Model/Country.cs:16]: space

File: Model/Country.cs [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    /// <summary>
+    /// AutoIncrement-ID of the config
+    /// </summary>
+    public int Id { get; set; }
+    /// <summary>
```
Comment (dawkstorm) [inline Model/Country.cs:22]: space missing

File: Model/Country.cs [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+    /// Is it a member of EUCU
+    /// </summary>
+    public bool IsEUCU { get; set; }
+    /// <summary>
+    /// Concstructor
+    /// </summary>
+    /// <param name="A2Code">Country's A2 isocode</param>
+    /// <param name="IsEUCU">Is it a member of EUCU</param>
+    public Country(string A2Code, bool IsEUCU)
```
Comment (dawkstorm) [inline Model/Country.cs]: constructor on top

File: Model/CountryConfiguration.cs [hunk 16]
```diff
@@ -0,0 +1,14 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+public class CountryConfiguration : IEntityTypeConfiguration<Country>
```
Comment (dawkstorm) [inline Model/CountryConfiguration.cs:7]: summary missing

File: Model/CustomsContext.cs [hunk 17]
```diff
@@ -0,0 +1,36 @@
+using Microsoft.EntityFrameworkCore;
+
+
```
Comment (dawkstorm) [inline Model/CustomsContext.cs]: remove space

File: Services/CustomsService.cs [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+            {
+                Console.WriteLine("country is not in the database");
+                throw;
+            }
+
+
+        }
+
+        //Check does specific postal code belong to EUCU
+        bool CheckForEUCU(string country, string pCode)
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: private

File: Model/CountryConfiguration.cs [hunk 19]
```diff
@@ -0,0 +1,20 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
```
Comment (dawkstorm) [inline Model/CountryConfiguration.cs:2]: space missing

File: Model/CustomsContext.cs [hunk 20]
```diff
@@ -0,0 +1,32 @@
+using Microsoft.EntityFrameworkCore;
+
+/// <summary>
+/// Database model for Customs
+/// </summary>
+public class CustomsContext : DbContext
+{
+    public CustomsContext(DbContextOptions<CustomsContext> options) : base(options)
+    {
+
```
Comment (dawkstorm) [inline Model/CustomsContext.cs:9]: remove space

File: Model/CustomsContext.cs [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+    }
```
Comment (dawkstorm) [inline Model/CustomsContext.cs]: space missing

File: Model/PostalCode.cs [hunk 22]
```diff
@@ -0,0 +1,41 @@
+/// <summary>
+/// Database model for postal codes
+/// </summary>
+public class PostalCode
+{
+
```
Comment (dawkstorm) [inline Model/PostalCode.cs]: remove space

File: Model/PostalCode.cs [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+    /// <summary>
+    /// Costructor
```
Comment (dawkstorm) [inline Model/PostalCode.cs]: typo Constructor

File: Model/PostalCode.cs [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+    /// <summary>
+    /// Postal code 
+    /// </summary>
+    public string Code { get; set; }
+
+    /// <summary>
+    /// Type of the postal code: region or postal code
+    /// </summary>
+    public string Type { get; set; }
+
```
Comment (dawkstorm) [inline Model/PostalCode.cs]: remove spaces

File: Repository/CustomRepository.cs
Comment (dawkstorm) [inline Repository/CustomRepository.cs:1]: delete

File: Services/CustomsService.cs [hunk 25]
```diff
@@ -0,0 +1,143 @@
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace CustomsController.Services
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: namespace CustomsController.Services;

File: README.md [hunk 26]
```diff
@@ -0,0 +1,3 @@
+# zoll
+
```
Comment (dawkstorm) [inline README.md:2]: create readme

File: README.md [hunk 27]
```diff
@@ -0,0 +1,15 @@
+# Customs control check
+
+Customs control check for European countries, whether in EU or not. This project is for RelaxDays GmbH and my work as praktikant, it uses C# with RESTAPI and Entity Framework core. It's primary purpose to automate check on customs control between 2 postal codes in different countries.
```
Comment (dawkstorm) [inline README.md]: This project was done during my school internship at RelaxDays GmbH

File: Controllers/CountryController.cs [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+    [HttpGet]
+    [Route("get-all-countries")]
+    public async Task<List<Country>> GetAllCountries()
+    {
+        return _customService.GetAllCountries();
+    }
+
+    [HttpGet]
+    [Route("get-country-information")]
+    public async Task<IActionResult> GetCountryInformation(string A2Code)
```
Comment (dawkstorm) [inline Controllers/CountryController.cs:101]: summary

File: Model/Country.cs [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+    /// </summary>
+    public string A2Code { get; set; } //e.g. "DE", "PL"
+
+    /// <summary>
+    /// Is it a member of EUCU
+    /// </summary>
+    public bool IsEUCU { get; set; }
+
+    /// <summary>
+    ///
```
Comment (dawkstorm) [inline Model/Country.cs]: summary

File: Model/CustomsResponse.cs [hunk 30]
```diff
@@ -0,0 +1,6 @@
+public class CustomsResponse
+{
+    public bool? IfCustomsInEUCU { get; set; }
```
Comment (dawkstorm) [inline Model/CustomsResponse.cs:9]: summary

File: Model/PostalCode.cs [hunk 31]
```diff
@@ -0,0 +1,50 @@
+using CustomsController.Model;
+
```
Comment (dawkstorm) [inline Model/PostalCode.cs]: remove space

File: Model/PostalCodeType.cs [hunk 32]
```diff
@@ -0,0 +1,9 @@
+namespace CustomsController.Model
+{
+    public enum PostalCodeType
```
Comment (dawkstorm) [inline Model/PostalCodeType.cs]: summary

File: Services/CustomsService.cs [hunk 33]
```diff
@@ -0,0 +1,289 @@
...
+using Microsoft.EntityFrameworkCore;
+using Microsoft.OpenApi.Models;
+
+namespace CustomsController.Services;
+
+/// <inheritdoc/>
+public class CustomsService : ICustomsService
+{
+    private readonly CustomsContext _customContext; // database itself
+    /// <summary>
```
Comment (dawkstorm) [inline Services/CustomsService.cs:10]: add space

File: Services/CustomsService.cs [hunk 34]
```diff
@@ -0,0 +1,289 @@
...
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.OpenApi.Models;
+
+namespace CustomsController.Services;
+
+/// <inheritdoc/>
+public class CustomsService : ICustomsService
+{
+    private readonly CustomsContext _customContext; // database itself
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: remove comment

File: Services/CustomsService.cs [hunk 35]
```diff
@@ -0,0 +1,289 @@
...
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    /// <param name="customsContext">for logic</param>
+    public CustomsService(CustomsContext customsContext)
+    {
+        _customContext = customsContext;
+    }
+
+    public CustomsService() { }
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: remove

File: Services/CustomsService.cs [hunk 36]
```diff
@@ -0,0 +1,289 @@
...
+    /// <inheritdoc/>
+    public string GetCountryEUCU(string A2Code)
+    {
+
+        var countrytest = _customContext.Countries
+            .FirstOrDefault(c => c.A2Code == A2Code);
+        var countrytest2 = _customContext.Countries
+            .Include(c => c.PostalCodes)
+            .FirstOrDefault(c => c.A2Code == A2Code);
+
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: remove space

File: Services/CustomsService.cs [hunk 37]
```diff
@@ -0,0 +1,289 @@
...
+    }
+
+    /// <inheritdoc/>
+    public CustomsResponse GetCustoms(string country1code, string country2code)
+    {
+
+        var country1_isEUCU = _customContext.Countries.FirstOrDefault(c => c.A2Code == country1code);
+        var country2_isEUCU = _customContext.Countries.FirstOrDefault(c => c.A2Code == country2code);
+
+        // var country1_isEUCU = bool.Parse(GetCountryEUCU(country1code));
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: remove comment

File: Services/CustomsService.cs [hunk 38]
```diff
@@ -0,0 +1,289 @@
...
+
+        var result2 = new CustomsResponse()
+        {
+            IfCustomsInEUCU = false,
+            Success = true
+        };
+        return result2;
+
+
+        // if (country1_isEUCU && country2_isEUCU) return false;
```
Comment (dawkstorm) [inline Services/CustomsService.cs]: remove space and comment

File: Services/ICustomsService.cs [hunk 39]
```diff
@@ -0,0 +1,68 @@
+namespace CustomsController.Services
```
Comment (dawkstorm) [inline Services/ICustomsService.cs]: namespace yzx; declaration

File: Services/ICustomsService.cs [hunk 40]
```diff
@@ -0,0 +1,68 @@
...
+        /// <param name="p1">Shipper postalCode</param>
+        /// <param name="c2">Receiver country</param>
+        /// <param name="p2">Receiver postalCode</param>
+        public CustomsResponse GetCustomsBetweenDistricts(
+            string shipperC1,
+            string shipperP1,
+            string receiverC2,
+            string receiverP2,
+            string city1 = "",
+            string city2 = "");
```
Comment (dawkstorm) [inline Services/ICustomsService.cs]: add summary

File: Controllers/CountryController.cs (hunk 28 above)
Comment (dawkstorm) [reply Controllers/CountryController.cs:101]: done

File: Model/CountryConfiguration.cs (hunk 19 above)
Comment (dawkstorm) [reply Model/CountryConfiguration.cs:2]: done

File: Model/CustomsResponse.cs (hunk 30 above)
Comment (dawkstorm) [reply Model/CustomsResponse.cs:9]: done

File: Model/PostalCode.cs (hunk 31 above)
Comment (dawkstorm) [reply Model/PostalCode.cs]: done

File: Model/PostalCode.cs (hunk 23 above)
Comment (dawkstorm) [reply Model/PostalCode.cs]: done

File: Model/PostalCode.cs (hunk 24 above)
Comment (dawkstorm) [reply Model/PostalCode.cs]: done

File: Model/PostalCodeType.cs (hunk 32 above)
Comment (dawkstorm) [reply Model/PostalCodeType.cs]: done

File: README.md (hunk 26 above)
Comment (dawkstorm) [reply README.md:2]: done

File: README.md (hunk 27 above)
Comment (dawkstorm) [reply README.md]: done

File: Services/CustomsService.cs (hunk 34 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/CustomsService.cs (hunk 33 above)
Comment (dawkstorm) [reply Services/CustomsService.cs:10]: done

File: Services/CustomsService.cs (hunk 35 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/CustomsService.cs (hunk 36 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/CustomsService.cs (hunk 37 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/CustomsService.cs (hunk 38 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/ICustomsService.cs (hunk 40 above)
Comment (dawkstorm) [reply Services/ICustomsService.cs]: done

File: Model/Country.cs [hunk 41]
```diff
@@ -0,0 +1,36 @@
...
+    public Country(string A2Code, bool IsEUCU)
+    {
+        this.A2Code = A2Code;
+        this.IsEUCU = IsEUCU;
+    }
+
+    /// <summary>
+    /// AutoIncrement-ID of the config
+    /// </summary>
+    public int Id { get; set; }
```
Comment (ermueller) [inline Model/Country.cs:20]: Question for all tables: Where is it defined that the Ids auto-increment? Is that the default setting or is a config missing?

File: Model/PostalCode.cs [hunk 42]
```diff
@@ -0,0 +1,49 @@
...
+    public PostalCode(string Code, PostalCodeType Type)
+    {
+        this.Code = Code;
+        this.Type = Type;
+    }
+
+    /// <summary>
+    /// AutoIncrement-ID of the config
+    /// </summary>
+    public int Id { get; set; }
```
Comment (ermueller) [inline Model/PostalCode.cs]: See other comment about Auto-increments

File: Model/PostalCodeType.cs (hunk 32 above)
Comment (ermueller) [reply Model/PostalCodeType.cs]: "Exception" is a pretty heavily used and well-defined word in programming, can we use a synonym that isn't a programming keyword?

File: Controllers/CountryController.cs [hunk 43]
```diff
@@ -0,0 +1,102 @@
...
+
+    /// <summary>
+    /// Returns all countries
+    /// </summary>
+    [HttpGet]
+    [Route("get-all-countries")]
+    public async Task<List<Country>> GetAllCountries()
+    {
+        return _customService.GetAllCountries();
+    }
```
Comment (dawkstorm) [inline Controllers/CountryController.cs:92]: add space

File: Doc/Was haben wir gemacht.md [hunk 44]
```diff
@@ -0,0 +1,71 @@
...
+
+---
+
+## Zollbestimmung
+
+- Wie funktioniert Zoll in Europa?
+
+![[image.png | 400]]
+
+- Was sind gute "Regeln"
```
Comment (dawkstorm) [inline Doc/Was haben wir gemacht.md]: add ?

File: Model/Country.cs (hunk 41 above)
Comment (dawkstorm) [reply Model/Country.cs:20]: it is the default setting

File: Model/CustomsContext.cs (hunk 20 above)
Comment (dawkstorm) [reply Model/CustomsContext.cs:9]: done

File: Model/CustomsResponse.cs [hunk 45]
```diff
@@ -0,0 +1,18 @@
+/// <summary>
+/// Custom response for CustomsController 
+/// </summary>
+public class CustomsResponse
+{
+    /// <summary>
+    /// Is country in EUCU?
+    /// </summary>
+    public bool? IfCustomsInEUCU { get; set; }
+    /// <summary>
```
Comment (dawkstorm) [inline Model/CustomsResponse.cs:11]: add space

File: Model/CustomsResponse.cs [hunk 46]
```diff
@@ -0,0 +1,18 @@
...
+    /// Were there any exceptions? true = no, false = yes
+    /// </summary>
+    public bool Success { get; set; }
+    /// <summary>
```
Comment (dawkstorm) [inline Model/CustomsResponse.cs:16]: add space

File: Model/PostalCode.cs (hunk 42 above)
Comment (dawkstorm) [reply Model/PostalCode.cs]: see above

File: Model/PostalCodeType.cs (hunk 32 above)
Comment (dawkstorm) [reply Model/PostalCodeType.cs]: SpecialCaseType

File: Model/PostalCodeType.cs [hunk 47]
```diff
@@ -0,0 +1,21 @@
...
+{
+    /// <summary>
+    /// Type of an exception
+    /// </summary>
+    public enum PostalCodeType
+    {
+        /// <summary>
+        /// Exception is a zipcode
+        /// </summary>
+        Zipcode,
```
Comment (dawkstorm) [inline Model/PostalCodeType.cs]: add space,
comment should be special case is a zipcode

File: Model/PostalCodeType.cs [hunk 48]
```diff
@@ -0,0 +1,21 @@
...
+        /// <summary>
+        /// Exception is a region
+        /// </summary>
+        Region,
+        /// <summary>
```
Comment (dawkstorm) [inline Model/PostalCodeType.cs]: add space

File: README.md [hunk 49]
```diff
@@ -0,0 +1,20 @@
+# Customs control check
+
+Customs control check for European countries, whether in EU or not. This project was done during my school internship at RelaxDays GmbH, it uses C# with RESTAPI and Entity Framework core. It's primary purpose to automate check on customs control between 2 postal codes in different countries.
```
Comment (dawkstorm) [inline README.md]: REST-API

Comment (dawkstorm) [reply README.md]: for European Union Customs Union (EUCU),

File: README.md [hunk 50]
```diff
@@ -0,0 +1,20 @@
...
+
+## Key contributions
+
+-   Designed endpoints and manager against existing rules and exceptions
+-   Use of EF core for work with databases (2 tables, Countries and PostalCodes)
+-   Logic for determining existence of customs control between 2 postal codes including the exceptions
+
+## Technical stack
+
+-   **Languages and frameworks:** C#, .NET Core, Entity Framework, WebAPI RESTAPI,
```
Comment (dawkstorm) [inline README.md]: remove , at the end
REST-API

File: Repository/CustomRepository.cs
Comment (dawkstorm) [inline Repository/CustomRepository.cs:1]: delete file

File: Services/ICustomsService.cs [hunk 51]
```diff
@@ -0,0 +1,68 @@
+namespace CustomsController.Services;
+/// <summary>
+/// Service for Customs
+/// </summary>
+public interface ICustomsService
```
Comment (dawkstorm) [inline Services/ICustomsService.cs:6]: add space

File: Model/CustomsResponse.cs (hunk 45 above)
Comment (dawkstorm) [reply Model/CustomsResponse.cs:11]: added

File: Model/CustomsResponse.cs (hunk 46 above)
Comment (dawkstorm) [reply Model/CustomsResponse.cs:16]: added

File: Services/CustomsService.cs [hunk 52]
```diff
@@ -0,0 +1,265 @@
...
+    /// Constructor
+    /// </summary>
+    /// <param name="customsContext">for logic</param>
+    public CustomsService(CustomsContext customsContext)
+    {
+        _customContext = customsContext;
+    }
+
+    /// <inheritdoc/>
+    public Country AddNewCountry(string A2Code, bool isEUCU)
```
Comment (ermueller) [inline Services/CustomsService.cs]: Just `AddCountry` is enough i think

File: Services/CustomsService.cs [hunk 53]
```diff
@@ -0,0 +1,265 @@
...
+    {
+        var country = new Country(A2Code, isEUCU);
```
Comment (ermueller) [inline Services/CustomsService.cs]: you can do the check before creating the country object

File: Services/CustomsService.cs [hunk 54]
```diff
@@ -0,0 +1,265 @@
...
+    public string GetCountryEUCU(string A2Code)
+    {
+        var country = _customContext.Countries.FirstOrDefault(c => c.A2Code == A2Code);
+        if (country != default)
+        {
+            return $"{country.IsEUCU}";
+        }
+        else
+        {
+            return "country is not in the database";
```
Comment (ermueller) [inline Services/CustomsService.cs]: Shouldn't this throw an Exception?

Comment (ermueller) [reply Services/CustomsService.cs]: NotFound

File: Services/CustomsService.cs [hunk 55]
```diff
@@ -0,0 +1,265 @@
...
+
+        var result2 = new CustomsResponse()
+        {
+            IfCustomsInEUCU = !(country1_isEUCU && country2_isEUCU),
+            Success = true
+        };
+        return result2;
+    }
+
+    private CustomsResponse DoesCountryExist(string country1code, string country2code)
```
Comment (ermueller) [inline Services/CustomsService.cs:100]: If we phrase the function name like a y/n-question we should also return a bool and not an object. You can either rename the function or change the response to a bool and work with that

File: Services/CustomsService.cs [hunk 56]
```diff
@@ -0,0 +1,265 @@
...
+        }
+        return true;
+    }
+
+    private bool ZipcodeCheck(string pCode, IQueryable<PostalCode> exceptionsForCountry)
+    {
+        // postleitzahl check
+        // if (exceptionCodes.Contains(pCode) )
+        //    return false; // customs ist true
+        foreach (var exceptionForCountry in exceptionsForCountry)
```
Comment (ermueller) [inline Services/CustomsService.cs:186]: I think this function is an optimal case for LINQ, but this is more of an FYI and not something you have to change 😄

File: Controllers/CustomsController.cs [hunk 57]
```diff
@@ -0,0 +1,58 @@
...
+    /// </summary>
+    /// <param name="country1">Shipper country's isocode</param>
+    /// <param name="p1">Shipper postalCode</param>
+    /// <param name="country2">Receiver country's isocode</param>
+    /// <param name="p2">Receiver postalCode</param>
+    /// <param name="city1">Shipper's city</param>
+    /// <param name="city2">Receiver's city</param>
+    [HttpGet]
+    [Route("get-customs-with-zipcodes")]
+    public async Task<CustomsResponse> GetCustomsWithZipcodes(
```
Comment (ermueller) [inline Controllers/CustomsController.cs:48]: This is a pretty complex input that would be more understandable by utilizing classes

File: Controllers/CustomsController.cs [hunk 58]
```diff
@@ -0,0 +1,58 @@
...
+        string country1,
+        string country2,
+        string p1,
+        string p2,
+        string city1 = "",
+        string city2 = "")
+    {
+        return _customService.GetCustomsBetweenDistricts(country1, p1, country2, p2, city1, city2);
```
Comment (ermueller) [inline Controllers/CustomsController.cs:56]: I think you can utilize classes to make this input more easily understandable - juggling 6 parameters with "1" and "2" at the end can become a bit confusing

File: Model/SpecialCaseType.cs [hunk 59]
```diff
@@ -0,0 +1,22 @@
...
+    /// </summary>
+    Zipcode,
+
+    /// <summary>
+    /// Special case is a region
+    /// </summary>
+    Region,
+
+    /// <summary>
+    /// Special case is a cit  y
```
Comment (dawkstorm) [inline Model/SpecialCaseType.cs]: city

File: Services/CustomsService.cs (hunk 53 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/CustomsService.cs (hunk 54 above)
Comment (dawkstorm) [reply Services/CustomsService.cs]: done

File: Services/CustomsService.cs (hunk 55 above)
Comment (dawkstorm) [reply Services/CustomsService.cs:100]: ok

File: Services/CustomsService.cs (hunk 56 above)
Comment (dawkstorm) [reply Services/CustomsService.cs:186]: var result = !exceptionsForCountry.Any(exceptionForCountry => exceptionForCountry.Code == pCode && exceptionForCountry.Type == SpecialCaseType.Zipcode);

File: Controllers/CustomsController.cs (hunk 58 above)
Comment (dawkstorm) [reply Controllers/CustomsController.cs:56]: ok

**Comment by dawkstorm** [comment]
tegshig approves

**Pull request merged**